scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1052,7 +1052,17 @@ def cmake(self, filter_stages=[]):
         return self.run_cmake(args,filter_stages)
 
     def build(self):
-        return self.run_build(['--build', self.build_dir])
+        harness = HarnessImporter.get_harness(self.instance.testsuite.harness.capitalize())
+        build_result = self.run_build(['--build', self.build_dir])
+        try:
+            harness.instance = self.instance
```

Comment:
nit: maybe better would be to call `harness.configure(instance)` as in `run` method, where harness is also created.